src/utils/WindowProvider.js

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) Microsoft Corporation and others. */
+/* SPDX-License-Identifier: MIT */
+
+import React, { PureComponent, createContext } from 'react'
```

Comment:
BTW, did this code come from somewhere or did you write it from scratch?  Seems quite different. uses `var` not `const`, `function` rather than `=>`, names like `obj`, `func` etc that don't fit in the overall coding style.

Reply:
I took the `throttle()` function from MDN:
https://developer.mozilla.org/en-US/docs/Web/Events/resize#requestAnimationFrame_customEvent
I'll update the syntax

Reply:
instead of taking the throttle from MDN, can we use lodash throttle?
They have updated docs for using a timeout throttle instead of requestAnimationFrame

https://developer.mozilla.org/en-US/docs/Web/Events/resize#setTimeout

I think it would look something like this:

```js
  componentDidMount() {
    window.addEventListener('resize', _.throttle(this.updateDimensions))
  }

  componentWillUnmount() {
    window.removeEventListener('resize')
  }
```

what do you think?

Reply:
sure. sounds good

Reply:
I've used lodash and use debounce since it makes more sense than throttle in this case